aten/src/ATen/native/Repeat.cpp

```diff
@@ -102,8 +102,17 @@ Tensor repeat_interleave(
     int64_t repeats,
     c10::optional<int64_t> dim,
     c10::optional<int64_t> output_size) {
+  Tensor input = self;
   at::Tensor repeats_ = at::empty(1, self.options().dtype(at::kLong)).fill_(repeats);
-  return at::native::repeat_interleave(self, repeats_, dim, output_size);
```

Comment:
Can you explain why you had to make these changes?

Reply:
> I also am not sure why you need the C++ change (or, if you are changing a composite, you need to work harder. Check https://docs.google.com/document/d/1GgvOe7C8_NVOMLOCwDaYV1mXXyHMXY7ExoewHqooxrs/edit# )

@ezyang Thanks for the info. Yes, `aten.repeat_interleave.self_int` is a composite. But I don't intend to reimplement a decomposition for it from the python side but just want to have it work with `meta` device. That's why I changed the C++ implementation to initialize the `output_size` parameter which is needed by `meta` implementation of `repeat_interleave.Tensor` that is called by `aten.repeat_interleave.self_int`. From the doc you shared, do I have to reimplement a decomposition under `torch/_decomp` for `aten.repeat_interleave.self_int` if I don't change C++?

Reply:
I guess my point is, it's better to go all the way and do a bit more, it will be about the same number of LOC changed but it will work in more cases.

What I am expecting to see is either:

1. The decomp is ported entirely to Python, or
2. The c++ decomp is symint'ified so that it is renamed to blah_symint, takes symint args and calls symint functions underneath

Reply:
Yes, exactly like `repeat_interleave_symint`. So actually, it seems like we already partially symintified this. You just need to make it more clever about when there is `output_size specified.

Reply:
The problem doesn't happen when "output_size" *is* specified. The meta device can work well with that (https://github.com/pytorch/pytorch/blob/main/torch/_meta_registrations.py#L1047). The problem is when "output_size" *is not* specified and the decomposition into `repeat_interleave.Tensor` makes it impossible to infer the output_size any longer. That's why I initialize the "output_size" in the composite before calling into `repeat_interleave.Tensor`. Any better suggestion?